misc/test/performance_test.go

```diff
@@ -177,6 +178,32 @@ func TestManyResources(t *testing.T) {
 	}
 }
 
+func programTestAsBenchmark(
+	t *testing.T,
+	bench traces.Benchmark,
+	test integration.ProgramTestOptions) {
+
+	// Run preview only to make sure all needed plugins are
+	// downloaded so that these downloads do not skew
+	// measurements.
+	t.Run("prewarm", func(t *testing.T) {
```

Comment:
Go question: does this guarantee prewarm will always run before "benchmark"? It seems to work but I was not sure. I did not want to introduce these blocks but ProgramTest does not like running twice in the same testing.T.

Reply:
I think only because they are not marked as parallel (the linter in pu/pu might complain about this actually).

Reply:
> ProgramTest does not like running twice in the same testing.T.

That's strange. Yeah, I think semantically it doesn't make sense to "prewarm" as a subtest. We can merge -- I don't think this code is super delicate, so if it works, it works. It's okay if it's less than perfect.

Reply:
It does seem to work on one example, but we can merge and let it run a few days and get data. This is not customer-facing.